File: app/models/gauge_value.rb
# Formats monitoring gauge readings for display.
#
# Fractional values are padded to +precision+ places (541.1 → "541.10").
# Whole numbers omit a trailing decimal (540 → "540"). Discharge-style
# precision 0 always renders as an integer.
module GaugeValue
  module_function

  def format(value, precision: 2, delimiter: ",")
    return if value.nil?

    precision = precision.to_i
    rounded = BigDecimal(value.to_s).round(precision)

    if precision <= 0 || rounded == rounded.to_i
      ActiveSupport::NumberHelper.number_to_delimited(rounded.to_i, delimiter: delimiter)
    else
      ActiveSupport::NumberHelper.number_to_rounded(
        rounded,
        precision: precision,
        delimiter: delimiter,
        strip_insignificant_zeros: false
      )
    end
  end
end

File: app/javascript/lib/gauge_value.js
// Formats monitoring gauge readings for display.
// Fractional values pad to `precision` places (541.1 → "541.10").
// Whole numbers omit a trailing decimal (540 → "540").
export function formatGaugeValue(value, precision = 2) {
  if (value == null || Number.isNaN(Number(value))) return null

  const digits = Number(precision)
  const fixed = Number(value).toFixed(digits)
  const numeric = Number(fixed)

  if (digits <= 0 || Number.isInteger(numeric)) {
    return Math.round(numeric).toLocaleString("en-US")
  }

  return numeric.toLocaleString("en-US", {
    minimumFractionDigits: digits,
    maximumFractionDigits: digits
  })
}
